modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AMQPMetaDataHandler.java

```diff
@@ -27,19 +27,30 @@
 import java.nio.ByteBuffer;
 
 /**
- * Will handle cloning of meta data at an even where AMQP message is published
+ * Will handle cloning of metadata at an even where AMQP message is published
  */
 public class AMQPMetaDataHandler {
 
-    public static byte[] constructMetadata(String routingKey, ByteBuffer buf, StorableMessageMetaData originalMeataData,
+    /**
+     * Construct metadata while creating a copy of MessageMetadata for AMQP; after an update of the routing key and
```

Comment:
Would it be better to summarize this ? more than construction of a new metadata object would it be more relevant to state that its updating existing metadata with the given set of values ?